internal/cloudinfo/providers/amazon/config.go

```diff
@@ -75,22 +61,40 @@ type Credentials struct {
 	// Shared credentials
 	SharedCredentialsFile string
 	Profile               string
+
+	// IAM role ARN to assume
+	AssumeRoleARN string
 }
 
 func configFromCredentials(creds Credentials) *aws.Config {
-	providers := []credentials.Provider{
-		&credentials.StaticProvider{Value: credentials.Value{
+	var providers []credentials.Provider
+
+	if creds.AccessKey != "" && creds.SecretKey != "" {
+		providers = append(providers, &credentials.StaticProvider{Value: credentials.Value{
 			AccessKeyID:     creds.AccessKey,
 			SecretAccessKey: creds.SecretKey,
 			SessionToken:    creds.SessionToken,
-		}},
-		&credentials.SharedCredentialsProvider{
+		}})
+	}
+
+	if creds.SharedCredentialsFile != "" {
+		providers = append(providers, &credentials.SharedCredentialsProvider{
 			Filename: creds.SharedCredentialsFile,
 			Profile:  creds.Profile,
-		},
+		})
+	}
+
+	awsConfig := &aws.Config{}
+	if len(providers) > 0 {
+		awsConfig.Credentials = credentials.NewChainCredentials(providers)
 	}
 
-	return &aws.Config{
-		Credentials: credentials.NewChainCredentials(providers),
+	if creds.AssumeRoleARN != "" {
+		sess := session.Must(session.NewSession(awsConfig))
```

Comment:
This is a bit problematic, because `session.Must` panics if it fails to read files or environment variables. Panics aren't great, so this function should return an error instead.

Reply:
Fixed in 44aa973f167ba73f18e4c0690567e9f4098836cf